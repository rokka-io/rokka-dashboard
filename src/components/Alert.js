import PropTypes from 'prop-types'
import React from 'react'
import cx from 'classnames'

const Alert = ({ alert = null }) => {
  if (!alert) {
    return null
  }

  let message = alert.message
  if (Array.isArray(message)) {
    message = message.map((msg, index) => (<span className="rka-alert-msg" key={index}>{msg}</span>))
  } else {
    message = (<span className="rka-alert-msg">{message}</span>)
  }

  return <div className={cx('rka-alert', `is-${alert.type}`, 'txt-c')}>{message}</div>
}
Alert.propTypes = {
  alert: PropTypes.shape({
    type: PropTypes.string,
    message: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)])
  })
}

export default Alert
