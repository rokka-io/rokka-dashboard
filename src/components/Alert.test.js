import React from 'react'
import Alert from './Alert'
import renderer from 'react-test-renderer'

test('Alert does not render if no alert props given', () => {
  const component = renderer.create(<Alert />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Alert renders one alert', () => {
  const component = renderer.create(<Alert alert={{ type: 'error', message: 'Test alert' }} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Alert renders multiple alerts', () => {
  const component = renderer.create(
    <Alert alert={{ type: 'error', message: ['Test alert', 'Rokka yay'] }} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
