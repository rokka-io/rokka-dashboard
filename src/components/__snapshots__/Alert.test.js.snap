// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render if no alert props given 1`] = `null`;

exports[`Alert renders multiple alerts 1`] = `
<div
  className="rka-alert is-error txt-c"
>
  <span
    className="rka-alert-msg"
  >
    Test alert
  </span>
  <span
    className="rka-alert-msg"
  >
    Rokka yay
  </span>
</div>
`;

exports[`Alert renders one alert 1`] = `
<div
  className="rka-alert is-error txt-c"
>
  <span
    className="rka-alert-msg"
  >
    Test alert
  </span>
</div>
`;
